Editor/UI/EditorWindows/SetupGuide/SetupGuide.cs

```diff
@@ -74,18 +74,29 @@ private VisualElement InitializeSubdomainPanel()
             {
                 nextButton.SetEnabled(!string.IsNullOrEmpty(subdomain));
             };
-
-            subdomainPanel.Q<Toggle>(USE_DEMO_SUBDOMAIN_TOGGLE).RegisterValueChangedCallback(x =>
+            if (!ProjectPrefs.GetBool(USE_DEMO_SUBDOMAIN_TOGGLE) && CoreSettingsHandler.CoreSettings.Subdomain == CoreSettings.DEFAULT_SUBDOMAIN)
```

Comment:
Wouldn't this break the flow for the users, if they open the Setup Guide and close it? Could we do something like this, that first time, the CoreSettings.Subdomain is set to empty string and we can check that it isn't empty. So for the users, who currently have the CoreSettings.Subdomain set to demo, it would stay as demo?

Reply:
That way, we basically don't need the USE_DEMO_SUBDOMAIN_TOGGLE

Reply:
Ahh, just realised, that you need to know somehow, that the subdomain toggle is switched on.

Reply:
Yep admittedly it is overly complicated logic. But it is done that way to match the product UX requirements discussed with Bernhard. Basically, this logic enforces the blocking of the next step if they haven't changed from demo && demo checkbox NOT checked.